app/models/work.rb

```diff
@@ -0,0 +1,45 @@
+class Work < ApplicationRecord
+  # self note: the dependent: :delete_all allows for a work to be deleted even though a user and a work are connected using foreign keys in the votes table
+  has_many :votes, dependent: :delete_all
+
+  validates :title, presence: true, uniqueness: true
+  validates :category, presence: true
+
+  # def find_work_id
+  #   return self.id
+  # end
+
+  # self.votes is returning the votes for the specific work
+  # the 'has_many :votes' is doing the work of finding votes whose work_id is equal to the work that is being passed in
+  def total_votes
+    num_of_votes = self.votes.count
+    return num_of_votes
+  end
+
+  def users_that_voted_on_this_work
+    # work_id = find_work_id
+    # array_of_votes = Vote.find_by(work_id: work_id)
+    # array_of_votes = self.votes
+    array_of_votes = self.votes.order(created_at: :desc)
+    users_ids = []
+
+    array_of_votes.each do |vote|
+      users_ids << vote.user_id
+    end
+
+    array_of_users = []
+    users_ids.each do |x|
+      user = User.find_by(id: x)
+      array_of_users << user
+    end
+
+    return array_of_users
+  end
+
+  def find_vote_where_work_upvoted_by_user(user_id)
```

Comment:
Even though this method name is long (almost too long), I think it's pretty descriptive... Maybe there's a way we can get this detailed name without making it this long (maybe `get_vote_by_user`?)